src/server/room-controller.spec.js

```diff
@@ -158,36 +159,133 @@ describe('room-controller', () => {
   });
 
   describe('handleGetRoomPage', () => {
-    const room = { roomId: 'roomId', name: 'Mein schöner Raum' };
+    const privateRoom = { roomId: 'roomId', name: 'Mein schöner Raum', owner: 'owner', access: ROOM_ACCESS_LEVEL.private };
+
+    beforeEach(() => {
+      clientDataMapper.mapRoom.resolves(privateRoom);
+
+      roomService.getRoomById.callsFake(roomId => {
+        if (roomId === privateRoom.roomId) {
+          return Promise.resolve(privateRoom);
+        }
+        return Promise.resolve();
+      });
+    });
+
+    describe('when the private room exists', () => {
+      describe('when the room owner calls', () => {
+
+        const request = {
+          params: {
+            roomId: 'roomId'
+          },
+          user: {
+            _id: 'owner'
+          }
+        };
+
+        const invitations = [{ email: 'lae@bucalae.com', sentOn: new Date() }];
+
+        beforeEach(async () => {
+          roomService.getRoomInvitations = sandbox.stub().resolves(invitations);
+          roomService.isRoomMemberOrOwner = sandbox.stub().resolves(true);
```

Comment:
I see stubbing is now implemented in 2 ways in this file.
You could use the already existing pattern and place all `roomService` stubs together with the others in the out-most `beforeEach` and then stub the return as in the other usages.

Example:
```
roomService.getRoomInvitations.resolves(invitations);
```

Reply:
Fixed, thanks.

Reply:
Can you please fix **all** such cases that are not streamlined?
Meaning everything with `= sandbox.stub()` in the inner `beforeEach` blocks? It's the same case for all, you don't need re-stubbing, just setting the resolve.

Reply:
Somehow I tried this before and it didn't work, now it does, I must have done something wrong back then, thanks.
I think it's fixed now.

Reply:
You probably confused the methods, cause one was already stubbed in the outer most `beforeEach` and one not. And of course it would not work for the one that was not.